speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/unitTestUtilitiesTests/ComparerUtilsTests.cpp

```diff
@@ -0,0 +1,153 @@
+#include <gtest/gtest.h>
+#include "GeneratorUtils.h"
+#include "ComparerUtils.h"
+
+/// @brief Tests whether CompareSegments will correctly validate it's input.
+TEST(ComparerUtilsTests, SegmentsLongerThanSegmentCount)
+{
+    TestSegments testSegments = GeneratorUtils::GenerateSegments();
+    EXPECT_THROW(ComparerUtils::CompareSegments(testSegments.NextSegments, testSegments.NextSegments, testSegments.NextSegmentsCount + 1, true),
+                 std::invalid_argument);
+    GeneratorUtils::DestroySegments(testSegments);
+}
+
+/// @brief Tests whether comparisons work correctly between default-initialized structs
+TEST(ComparerUtilsTests, EmptyCompare)
+{
+    tCarElt car1 = {};
+    tCarElt car2 = {};
+    tSituation situation1 = {};
+    tSituation situation2 = {};
+    TestSegments testSegments1 = {};
+    TestSegments testSegments2 = {};
+    constexpr int segmentCountTest = 10;
+    tTrackSeg* segments1 = new tTrackSeg[segmentCountTest];
+    tTrackSeg* segments2 = new tTrackSeg[segmentCountTest];
```

Comment:
This can be changed to use a define for segmentCountTest and use that define to create stack allocated arrays